### PR TITLE
Add support for Premiere styled EDLs to the CMX 3600 adapter

### PR DESCRIPTION
**Pull request by douglascomet**
```Fixes```

[[1178](https://github.com/PixarAnimationStudios/OpenTimelineIO/issues/1178)] Premiere style for cmx3600

**Summary.**

Adds support for Premiere style for cmx3600 to limit the auto added comments that are added per clip when writing out an EDL.

**Tests.**

No new tests were added but the below is failing when running `make test` yet nearest_valid_timecode_rate seem like it should be available from the C++ class definition:

image

Additionally, when I attempt to run `make lint` I get this error even though `pydocstyle `is in my venv:
image

### CODE REVIEW

File: src/py-opentimelineio/opentimelineio/adapters/cmx_3600.py [hunk 1]
```diff
@@ -1377,14 +1384,13 @@ def _generate_comment_lines(
...
-            from_or_to=from_or_to,
-            url=url
-        ))
-    if url and style == 'nucoda':
-        lines.append("* {from_or_to} FILE: {url}".format(
+
+    # If the style has a spec, apply it and add it as a comment
+    style = VALID_EDL_STYLES.get(style)
+    if url and style:
+        lines.append("* {from_or_to} {style}: {url}".format(
```
Comment (reinecke) [inline src/py-opentimelineio/opentimelineio/adapters/cmx_3600.py:1391]: I love the DRYness of this change!

**Review by reinecke — APPROVED** [review]
I think we could merge this and be fine as is, I just had one minor thought. Thanks for the fix!

**Comment by reinecke** [comment]
Oops, sorry, one other note - would you mind adding a unittest to check the behavior?
Here is an [example](https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/2c374ffda97ad0aba799f8a624eaa3b20c7cb0cd/tests/test_cmx_3600_adapter.py#L713) of the unittest for the nucoda style:

File: src/py-opentimelineio/opentimelineio/adapters/cmx_3600.py [hunk 2]
```diff
@@ -86,9 +86,16 @@ class EDLParseError(exceptions.OTIOError):
...
+#   'premiere': '' (If Adobe Premiere's imports an EDL that uses
+#                   a "FROM" comment will cause the clips
+#                   to be name UNKNOWN)
 # When adding a new style, please be sure to add sufficient tests
 # to verify both the new and existing styles.
-VALID_EDL_STYLES = ['avid', 'nucoda']
+VALID_EDL_STYLES = {
+    'avid': 'CLIP',
+    'nucoda': 'FILE',
+    'premiere': '',
```
[the remark this answers is not in the document]
Comment (douglascomet) [reply src/py-opentimelineio/opentimelineio/adapters/cmx_3600.py:97]: Sure, easy enough change

**Comment by meshula** [comment]
@douglascomet This looks ready to merge, did you have any thoughts about @reinecke's request for a test?

**Review by reinecke — APPROVED** [review]
Looks good!

**Comment by apetrynet** [comment]
Hi, @douglascomet !
Seems like this PR has been stuck on a failing test run for some time. In fact I'm unable to re-run it. 
Would you mind re-basing your branch and pushing the updates again for the tests to update?

**Comment by douglascomet** [comment]
@apetrynet, thanks to your help we have resolved all the CI issues this PR was having!

**Comment by apetrynet** [comment]
Thank you for your contribution @douglascomet !

**Pull request merged**